features/admin.logs.v1/components/remote-logging-config-form.tsx

```diff
@@ -0,0 +1,461 @@
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Button from "@oxygen-ui/react/Button";
+import Card from "@oxygen-ui/react/Card";
+import Grid from "@oxygen-ui/react/Grid";
+import Skeleton from "@oxygen-ui/react/Skeleton";
+import Stack from "@oxygen-ui/react/Stack";
+import { AlertInterface, AlertLevels, IdentifiableComponentInterface } from "@wso2is/core/models";
+import { addAlert } from "@wso2is/core/store";
+import { CheckboxFieldAdapter, FinalForm, FinalFormField, FormValue, TextFieldAdapter } from "@wso2is/form/src";
+import {
+    ConfirmationModal,
+    DangerZone,
+    DangerZoneGroup,
+    Heading
+} from "@wso2is/react-components";
+import { FormValidation } from "@wso2is/validation";
+import { AxiosError } from "axios";
+import startCase from "lodash-es/startCase";
+import toLower from "lodash-es/toLower";
+import React, { ReactElement, useEffect, useState } from "react";
+import { FormRenderProps } from "react-final-form";
+import { useTranslation } from "react-i18next";
+import { useDispatch } from "react-redux";
+import { Dispatch } from "redux";
+import restoreRemoteLogPublishingConfiguration from "../api/restore-remote-log-publishing-configuration";
+import updateRemoteLogPublishingConfiguration from "../api/update-remote-log-publishing-configuration";
+import useTestRemoteLogPublishingConfiguration from "../api/use-test-log-publishing-configuration";
+import { LogType, RemoteLogPublishingConfigurationInterface } from "../models/remote-log-publishing";
+import "./remote-logging-config-form.scss";
+
+
+/**
+ * Props interface of {@link RemoteLoggingConfigForm}
+ */
+export interface RemoteLoggingConfigFormProps extends IdentifiableComponentInterface {
+    /**
+     * Log type such as `AUDIT`, `DEBUG`, etc.
+     */
+    logType: LogType;
+    /**
+     * Remote logging configuration such as `remoteUrl`, `connectTimeoutMillis`, etc.
+     */
+    logConfigData: RemoteLogPublishingConfigurationInterface;
+    /**
+     * Remote log publishing configs loading.
+     */
+    isLoading: boolean,
+    /**
+     * Errpr fetching log config.
+     */
+    error: AxiosError,
+    /**
+     * Callback to mutate the remote logging request.
+     */
+    mutateRemoteLoggingRequest: () => void;
+}
+
+/**
+ * Component to hold the remote logging configurations.
+ *
+ * @param props - Props injected to the component.
+ * @returns Remote logging component.
+ */
+export const RemoteLoggingConfigForm = ({
+    logType,
+    logConfigData,
```

Comment:
Shall we rename this to `initialValues`?
This won't cause confusion with `logConfig` variable inside the component. And makes the logics more readable.